File: server/internal/catalog/catalog.go
// Package catalog memuat aturan bisnis konversi satuan & katalog obat
// (dari README design handoff). qty obat SELALU disimpan dalam base unit.
package catalog

// Unit = satuan kemasan untuk menambah stok (mis. "Box (100)" -> mult 100).
type Unit struct {
	Label string `json:"label"`
	Mult  int    `json:"mult"`
}

// Rule = aturan per kategori obat: base unit, satuan kemasan, & ambang menipis.
type Rule struct {
	Base  string `json:"base"`
	Units []Unit `json:"units"`
	Thr   int    `json:"thr"`
}

// CAT: aturan per kategori. Kunci = kategori obat.
var CAT = map[string]Rule{
	"Tablet": {Base: "butir", Units: []Unit{{"Box (100)", 100}, {"Strip (10)", 10}, {"Butir", 1}}, Thr: 20},
	"Sirup":  {Base: "botol", Units: []Unit{{"Botol", 1}}, Thr: 5},
	"Sachet": {Base: "sachet", Units: []Unit{{"Box (100)", 100}, {"Sachet", 1}}, Thr: 20},
}

func rule(cat string) Rule {
	if r, ok := CAT[cat]; ok {
		return r
	}
	return CAT["Tablet"]
}

// BaseUnit -> base unit (butir/botol/sachet) untuk kategori.
func BaseUnit(cat string) string { return rule(cat).Base }

// Threshold -> ambang menipis untuk kategori.
func Threshold(cat string) int { return rule(cat).Thr }

// IsLow: menipis bila qty <= threshold.
func IsLow(cat string, qty int) bool { return qty <= Threshold(cat) }

// IsDanger (merah): qty <= threshold*0.5, selain itu warning (amber).
func IsDanger(cat string, qty int) bool { return float64(qty) <= float64(Threshold(cat))*0.5 }

// Item = entri katalog master obat untuk fitur "Cari Obat".
type Item struct {
	Name string `json:"name"`
	Cat  string `json:"cat"`
}

// CATALOG sengaja kosong. Dulu ia memuat 14 obat contoh dari design handoff,
// dan itu keliru: daftar obat adalah DATA milik klinik, bukan aturan bisnis.
// Obat karangan yang tidak pernah dimiliki klinik ikut muncul di pencarian dan
// tidak bisa dihapus tanpa deploy ulang.
//
// Sumber daftar obat sekarang tabel `medicines` — apa yang benar-benar
// distok bidan. Lihat PickModal.vue.
var CATALOG = []Item{}

// ValidCat: kategori yang dikenal. Dipakai sebelum menulis ke kolom `cat`,
// yang bertipe ENUM di MySQL — nilai di luar daftar ditolak database dengan
// pesan yang tidak menjelaskan apa-apa bagi pemakainya.
func ValidCat(cat string) bool {
	_, ok := CAT[cat]
	return ok
}

File: web/src/lib/catalog.js
// Aturan bisnis konversi satuan & katalog obat (dari README design handoff).
// qty obat SELALU disimpan dalam base unit. Mirror dari server Go (catalog.go).

export const CAT = {
  Tablet: { base: 'butir',  units: [ { label: 'Box (100)', mult: 100 }, { label: 'Strip (10)', mult: 10 }, { label: 'Butir', mult: 1 } ], thr: 20 },
  Sirup:  { base: 'botol',  units: [ { label: 'Botol', mult: 1 } ], thr: 5 },
  Sachet: { base: 'sachet', units: [ { label: 'Box (100)', mult: 100 }, { label: 'Sachet', mult: 1 } ], thr: 20 },
}

export const baseUnit = (cat) => (CAT[cat] || CAT.Tablet).base
export const threshold = (cat) => (CAT[cat] || CAT.Tablet).thr

// danger (merah) bila qty <= threshold*0.5, selain itu warning (amber)
export const isLow = (cat, qty) => qty <= threshold(cat)
export const isDanger = (cat, qty) => qty <= threshold(cat) * 0.5

// Katalog master obat untuk fitur "Cari Obat"
export const CATALOG = [
  { name: 'Paracetamol 500mg', cat: 'Tablet' },
  { name: 'Paracetamol sirup', cat: 'Sirup' },
  { name: 'Amoxicillin 500mg', cat: 'Tablet' },
  { name: 'Amoxicillin sirup', cat: 'Sirup' },
  { name: 'Ibuprofen 400mg', cat: 'Tablet' },
  { name: 'Asam Mefenamat 500mg', cat: 'Tablet' },
  { name: 'Cetirizine 10mg', cat: 'Tablet' },
  { name: 'CTM 4mg', cat: 'Tablet' },
  { name: 'Antasida', cat: 'Tablet' },
  { name: 'Vitamin B Complex', cat: 'Tablet' },
  { name: 'Vitamin C 500mg', cat: 'Tablet' },
  { name: 'Domperidone 10mg', cat: 'Tablet' },
  { name: 'Dexamethasone 0.5mg', cat: 'Tablet' },
  { name: 'Oralit', cat: 'Sachet' },
]

// Tint kategori untuk avatar/kotak (bg + warna teks). var(), bukan hex — lihat
// alasannya di komentar COL pada stores/klinik.js.
export const catTint = (cat) =>
  cat === 'Sirup' ? { bg: 'var(--sirup-tint)', ink: 'var(--sirup-ink)' }
  : cat === 'Sachet' ? { bg: 'var(--sachet-tint)', ink: 'var(--sachet-ink)' }
  : { bg: 'var(--tablet-tint)', ink: 'var(--tablet-ink)' }
